src/components/AnnotationShareControl.tsx

```diff
@@ -0,0 +1,168 @@
+   * It includes the result of copying to the clipboard, in case it failed.
+   */
+  onCopy?: (result: { successful: boolean }) => void;
+};
+
+function selectionOverflowsInputElement() {
+  // On iOS the selection overflows the input element
+  // See: https://github.com/hypothesis/client/pull/2799
+  return isIOS();
+}
+
+/**
+ * "Popup"-style component for sharing a single annotation.
+ *
+ * @param {AnnotationShareControlProps} props
+ */
+function AnnotationShareControl({
+  annotation,
+  group,
+  shareUri,
+  onCopy,
+}: AnnotationShareControlProps) {
+  const annotationIsPrivate = isPrivate(annotation.permissions);
+  const inContextAvailable = /^http(s?):/i.test(annotation.uri);
+  const inputRef = useRef<HTMLInputElement | null>(null);
+  const shareRef = useRef<HTMLDivElement | null>(null);
+
+  const [isOpen, setOpen] = useState(false);
+  const wasOpen = useRef(isOpen);
+
+  const toggleSharePanel = () => setOpen(!isOpen);
+  const closePanel = () => setOpen(false);
+
+  useEffect(() => {
+    if (wasOpen.current !== isOpen) {
+      wasOpen.current = isOpen;
+
+      if (isOpen && !selectionOverflowsInputElement()) {
+        // Panel was just opened: select and focus the share URI for convenience
+        inputRef.current?.focus();
+        inputRef.current?.select();
+      }
+    }
+  }, [isOpen]);
+
+  const copyShareLink = useCallback(async () => {
+    try {
+      await copyPlainText(shareUri);
+      onCopy?.({ successful: true });
+    } catch {
+      onCopy?.({ successful: false });
+    }
+  }, [onCopy, shareUri]);
+
+  if (!group) {
+    // This can happen if groups have just been unloaded but annotations have
+    // not yet been unloaded, e.g. on logout if a private group was focused
+    return null;
+  }
+
+  // Generate some descriptive text about who may see the annotation if they
+  // follow the share link.
+  // First: Based on the type of the group the annotation is in, who would
+  // be able to view it?
+  const groupSharingInfo =
+    group.type === 'private' ? (
+      <span>
+        Only members of the group <em>{group.name}</em> may view this
+        annotation.
+      </span>
+    ) : (
+      <span>Anyone using this link may view this annotation.</span>
+    );
+
+  // However, if the annotation is marked as "only me" (`annotationIsPrivate` is `true`),
+  // then group sharing settings are irrelevant—only the author may view the
+  // annotation.
+  const annotationSharingInfo = annotationIsPrivate ? (
+    <span>Only you may view this annotation.</span>
+  ) : (
+    groupSharingInfo
+  );
+
+  return (
+    // Make the container div focusable by setting a non-null `tabIndex`.
+    // This prevents clicks on non-focusable contents from "leaking out" and
+    // focusing a focusable ancestor. If something outside the panel gains
+    // focus, Popover's internal logic will close the panel.
+    // "Catch focus" here to prevent this.
+    // See https://github.com/hypothesis/client/issues/5196
```

Comment:
We should keep this as-is from the client in this PR, but reading the notes in the linked issue, this _might_ be obsolete when using a native popover.

Reply:
True! I misread it.